docs/tsm/TSM_FORMAT.md

```diff
@@ -0,0 +1,161 @@
+# TSM Inverted Index Proposal
+
+This is a proposal for adding a disk-based, inverted index for measurements, tags and values.  This is a extension and/or alternative of the ideas proposed in [TSI Proposal #7174](https://github.com/influxdata/influxdb/pull/7174).
+
+The inverted index supports the following:
+
+* Hash lookups of measurement, tags, and values
+* Sorted iteration over measurement, tags and values
+* Efficient series boolean operations over compressed bitmaps
+* Compressed series keys
+
+The index is a separate set of files specific to a shard that are compacted and merged periodically.  Searches run across multiple files and are merged.
+
+## Format
+
+┌────────────────────────────────────────────────────────────────┐
+│                         Inverted Index                         │
+├────────────────────┬──────────────────┬────┬───────────────────┤
+│    Series Dict     │     Tag Sets     │... │   Measurements    │
+└────────────────────┴──────────────────┴────┴───────────────────┘
+
+The `Inverted Index` contains three sections: Series Dictionary, Tag Sets and Measurements.  The `Series Dictionary` contains a dictionary encoded version of all series in the index.  The `Tag Sets` are the unique tag/value pairs for a given measurement.  The `Measurements` are the unique set of measurements in the index.
+
+┌───────────────────────────────────────────────────────┐
+│                   Series Dictionary                   │
+├───────────────────────────┬───────────────────────────┤
+│        Dictionary         │          Series           │
+├─────────┬─────────────────┼─────────┬─────────────────┤
+│   Len   │  Sorted Terms   │   Len   │  Sorted Series  │
```

Comment:
The terms would be sorted by measurement, then tag key so that their positions are smaller and encode in fewer bytes.  Since every series has measurement and most have a few tag keys, it would allow them to compress better.